src/raftstore/store/peer.rs

```diff
@@ -598,9 +602,26 @@ impl Peer {
         Ok(())
     }
 
+    fn has_leader_lease(&self) -> bool {
+        // check whether current peer is raft leader in its lease
+        if !self.raft_group.raft.in_lease() {
+            return false;
+        }
+        // check the clock time against clock drift bound
+        let lower_clock_drift_bound =
+            (self.raft_group.raft.randomized_election_timeout as f64 *
+             (1.0 - self.election_timeout_clock_drift_bound_percentage)) as usize;
+        self.raft_group.raft.election_elapsed < lower_clock_drift_bound
```

Comment:
here mean that 90% read in a election_timeout can use lease read, and the left 10% will go through raft read.